python/device/device.py

```diff
@@ -646,37 +756,100 @@ def check_for_failure_thread(self, worker_thread_info):
 
         while not self.done.isSet():
             worker_thread_info.watchdog_epochtime = time.time()
+            if self.is_paused():
+                time.sleep(1)
+                continue
 
             arrival_failure_count = 0
+            reported_properties_add_failure_count = 0
+            reported_properties_remove_failure_count = 0
             now = time.time()
+
             with self.pingback_list_lock:
-                for message_waiting in self.pingback_wait_list.values():
-                    if (
-                        now - message_waiting.send_epochtime
+                for pingback_wait_info in self.pingback_wait_list.values():
+                    if (pingback_wait_info.pingback_type == PingbackType.TELEMETRY_PINGBACK) and (
+                        now - pingback_wait_info.send_epochtime
                     ) > self.config.send_message_arrival_failure_interval_in_seconds:
                         logger.warning(
                             "arrival time for {} of {} seconds is longer than failure interval of {}".format(
-                                message_waiting.pingback_id,
-                                (now - message_waiting.send_epochtime),
+                                pingback_wait_info.pingback_id,
+                                (now - pingback_wait_info.send_epochtime),
                                 self.config.send_message_arrival_failure_interval_in_seconds,
                             )
                         )
                         arrival_failure_count += 1
 
+                    elif (
+                        (
+                            pingback_wait_info.pingback_type
+                            == PingbackType.ADD_REPORTED_PROPERTY_PINGBACK
+                        )
+                        and (now - pingback_wait_info.send_epochtime)
+                        > self.config.reported_properties_update_interval_in_seconds
+                    ):
+                        logger.warning(
+                            "reported property set time for {} of {} seconds is longer than failure interval of {}".format(
+                                pingback_wait_info.pingback_id,
+                                (now - pingback_wait_info.send_epochtime),
+                                self.config.reported_properties_update_interval_in_seconds,
+                            )
+                        )
+                        reported_properties_add_failure_count += 1
+
+                    elif (
+                        (
+                            pingback_wait_info.pingback_type
```

Comment:
Is this newline just because of line length?

Reply:
yes, I think so, but formatting decisions are made by the pretty-printer that we use.  I write code that is syntactically correct, and the pretty-printer makes it look consistent.  Sometimes it comes out ugly, but the rules are simple and consistent and mostly accepted as "the right way" by the community.